x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -191,3 +202,110 @@ func (k Keeper) deductProofSubmissionFee(ctx context.Context, supplierOperatorAd
 
 	return nil
 }
+
+// proofRequirementForClaim checks if a proof is required for a claim.
```

Comment:
Does this even need to be exposed?

```suggestion
// ProofRequirementForClaim checks if a proof is required for a claim.
```

Reply:
Yes, it's used in both `proof` and `tokenomics` keepers